src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Formatters/CollectionFormatter.cs

```diff
@@ -488,6 +491,42 @@ protected override ICollection<T> Complete(T[] intermediateCollection)
         }
     }
 
+    public sealed class InterfaceListFormatter2<T> : CollectionFormatterBase<T, List<T>, IList<T>>
+    {
+        protected override void Add(List<T> collection, int index, T value, MessagePackSerializerOptions options)
+        {
+            collection.Insert(index, value);
```

Comment:
use Add instead of Insert

Reply:
I went with `Insert` because an `index` is provided to this method. But if we're ok assuming that this should always append, I can call `Add` instead.